#11247: Remove in-place flag in binary operations

mouliraj-mcw opened the pull request:
Ticket : #11247 

### Work Done:

- Removed the in-place flags in binary operations.
- Updated the structure for in-place ops and added the tests.

### Checklist
- [x] [All Post commit CI](https://github.com/tenstorrent/tt-metal/actions/runs/10455081766)
- [ ] Model Perf Regression

Commit:
#11247: Remove in-place flag in binary operations

KalaivaniMCW commented on tests/ttnn/unit_tests/operations/test_add.py:
```diff
@@ -259,7 +259,7 @@ def test_prim_add(device, shape):
     input_tensor_a = ttnn.from_torch(torch_input_tensor_a, layout=ttnn.TILE_LAYOUT, device=device)
     input_tensor_b = ttnn.from_torch(torch_input_tensor_b, layout=ttnn.TILE_LAYOUT, device=device)
-    output_tensor = ttnn.prim.binary(input_tensor_a, input_tensor_b, ttnn.BinaryOpType.ADD, in_place=True)
+    output_tensor = ttnn.prim.binary(input_tensor_a, input_tensor_b, ttnn.BinaryOpType.ADD)
```
need to find a way to use output_tensor arg and perform inlace instead, since we are passing BinaryOpType directly

KalaivaniMCW reviewed (approved):
Approved.

eyonland reviewed (changes requested):
Please add unit tests.

mouliraj-mcw commented:
Status: Will add Test unit test and then merge.

Commit:
#11247: Add unit test

dmakoviichuk-tt commented:
What is the current status of inplace now for binary op?

dmakoviichuk-tt commented:
What I see:
new function ttnn.add_ and new tests for it.
What PR says: "remove inpalce flag in binaru operations"
@mouliraj-mcw  please always add in the description all new registered functions.
Also there should be added any changes to the api. 
Unfortunately right now it is impossible to track changes.

mouliraj-mcw commented:
> What I see: new function ttnn.add_ and new tests for it. What PR says: "remove inpalce flag in binaru operations" @mouliraj-mcw please always add in the description all new registered functions. Also there should be added any changes to the api. Unfortunately right now it is impossible to track changes.

Hi @dmakoviichuk-tt .

- The ttnn.add_ and other in-place operations were already available. I updated the structure by removing the in-place flag, and since it's an in-place version, I added the required unit tests.
- Let me Update the Description Clearly. Thankyou.
